Content.Server/GameObjects/Components/Items/RCDComponent.cs

```diff
@@ -0,0 +1,169 @@
+        [Dependency] private readonly IMapManager _mapManager;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager;
+        [Dependency] private IServerNotifyManager _serverNotifyManager;
+#pragma warning restore 649
+        public override string Name => "RCD";
+        private string _outputTile = "floor_steel";
+        private RCDmode mode = 0; //What mode are we on? Can be floors, walls, deconstruct.
+        private Array modes = Enum.GetValues(typeof(RCDmode));
+        private int ammo = 5; //How much "ammo" we have left. You can refille this with RCD ammo.
+
+        ///Enum to store the different mode states for clarity.
+        private enum RCDmode
+        {
+            Floors,
+            Walls,
+            Deconstruct
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _outputTile, "output", "floor_steel");
+        }
+
+
+        ///Method called when the RCD is clicked in-hand, this will swap the RCD's mode from "floors" to "walls".
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
+        {
+            SwapMode(eventArgs);
+            return true;
+        }
+
+        /**
+         *
+         * Method to allow the user to swap the mode of the RCD by clicking it in hand, the actual in hand clicking bit is done over on UseEntity()
+         *
+         * @param UseEntityEventArgs = The entity which triggered this method call, used to know where to play the "click" sound.
+         *
+         */
+
+        public void SwapMode(UseEntityEventArgs eventArgs)
+        {
+            _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/genhit.ogg", Owner);
+            int mode = (int) this.mode; //Firstly, cast our RCDmode mode to an int (enums are backed by ints anyway by default)
+            mode = (++mode) % modes.Length; //Then, do a rollover on the value so it doesnt hit an invalid state
+            this.mode = (RCDmode) mode; //Finally, cast the newly acquired int mode to an RCDmode so we can use it.
+            switch (this.mode)
+            {
+                case RCDmode.Floors:
+                    _outputTile = "floor_steel";
+                    break;
+                case RCDmode.Walls:
+                    _outputTile = "base_wall";
+                    break;
+                case RCDmode.Deconstruct:
+                    _outputTile = "space";
+                    break;
+            }
+            _serverNotifyManager.PopupMessage(Owner, eventArgs.User, "The RCD is now set to "+this.mode+" mode."); //Prints an overhead message above the RCD
+        }
+
+        /**
+         *
+         * Method called when the user examines this object, it'll simply add the mode that it's in to the object's description
+         * @params message = The original message from examining, like ..() in BYOND's examine
+         *
+         */
```

Comment:
You should use the [c# standard for documentation](https://docs.microsoft.com/en-us/dotnet/csharp/codedoc) instead.